tests/components/melcloud/test_atw_zone_sensor.py

```diff
@@ -0,0 +1,41 @@
+"""Test the MELCloud ATW zone sensor."""
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.melcloud.sensor import AtwZoneSensor
+
+
+@pytest.fixture
+def mock_device():
+    """Mock MELCloud device."""
+    with patch("homeassistant.components.melcloud.MelCloudDevice") as mock:
+        mock.name = "name"
+        mock.device.serial = 1234
+        mock.device.mac = "11:11:11:11:11:11"
+        yield mock
+
+
+@pytest.fixture
+def mock_zone_1():
+    """Mock zone 1."""
+    with patch("pymelcloud.atw_device.Zone") as mock:
+        mock.zone_index = 1
+        yield mock
+
+
+@pytest.fixture
+def mock_zone_2():
+    """Mock zone 2."""
+    with patch("pymelcloud.atw_device.Zone") as mock:
+        mock.zone_index = 2
+        yield mock
+
+
+def test_zone_unique_ids(mock_device, mock_zone_1, mock_zone_2):
+    """Test unique id generation correctness."""
+    sensor_1 = AtwZoneSensor(mock_device, mock_zone_1, "room_temperature", {})
```

Comment:
We should set up the integration and assert the entity unique_id via the entity registry. We shouldn't interact with the entity instance directly.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations